BrainPortal/app/views/tool_configs/_form_fields.html.erb

```diff
@@ -22,172 +21,187 @@
+      <% end %>
+
+      <% t.edit_cell(:version_name, :header => "Version", td_options: {:title => "Version"}) do |f| %>
+        <%= f.text_field :version_name %>
+        <div class="field_explanation"> Must be a simple string that represent
+        a short identifier for the version. First character must be alphanum,
+        and can contain only alphanums, \'.\', \'-\', \'_\', \':\' and \'@\' </div>
+      <% end %>
+
+      <% t.edit_cell(:description, :header => "Description", :content => full_description(@tool_config.description),
+                                     td_options: {:title => "Brief description for this configuration"}) do |f| %>
+        <%= f.text_area :description, :rows => 10, :cols => 40 %>
+        <div class="field_explanation">The first line must be a short summary, and the rest are for any special
+         notes for the users.</div>
+      <% end %>
+
+      <% t.edit_cell(:ncpus,
+                     :header => "Suggested CPUs per task",
+                     :td_options => {:title => "Suggested number of CPUs to use in parallel, per task"}) do |f| %>
+        <%= f.select :ncpus, [1, 2, 3, 4, 5, 6, 7, 8, 10, 12, 14, 16, 20, 24, 32, 36, 40, 48, 56, 64, 128, 192, 256,
+                              320, 384, 512] %>
+      <% end %>
+    <% end %>
   <% end %>
-
-  <% context = @tool_config.tool_id ? "tool" : "Execution Server" %>
-  <span title="Environment variables needed for this <%= context %>">
-    <p><label>Environment variables needed for this <%= context %>:</label><br/>
+  <%= show_table(@tool_config, :form_helper => cf, :header => "Environment variables",
+                               :edit_condition => check_role(:admin_user)) do |t| %>
     <% @tool_config.env_array.each do |env_name_value| %>
-      <% env_name  = env_name_value[0] %>
+      <% env_name = env_name_value[0] %>
       <% env_value = env_name_value[1] %>
-      <%= render :partial => 'env_key_value_pair', :locals => { :env_name => env_name, :env_value => env_value } %>
+      <% render :partial => 'env_key_value_pair_2', :locals => {:env_name => env_name,
+                                                                :env_value => env_value,
+                                                                :t => t
+      }
+      %>
     <% end %>
     <% 1.upto(5) do %>
-      <%= render :partial => 'env_key_value_pair', :locals => { :env_name => "", :env_value => "" } %>
+      <% render :partial => 'env_key_value_pair_2', :locals => {:env_name => "",
+                                                                :env_value => "",
+                                                                :t => t
+      } %>
+    <% end %>
+    <% t.edit_cell(:bourreau_id, :header => '',:content => '') {|f| f.hidden_field :bourreau_id } %>
+    <% t.edit_cell(:tool_id, :header => '', :content => '', :td_options => {:style => 'empty-cells: hide;'}) do |f| %>
+      <%= f.hidden_field :tool_id %>
```

Comment:
hidden variable wrong spot 

also

info message to move to bottom left of show table